src/System.Azure.Experimental/System/Azure/Sha256.Windows.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace System.Buffers.Cryptography
+{
+    public struct Sha256
+    {
+        IntPtr _algorithm;
+        IntPtr _hash;
+
+        public static Sha256 Create(byte[] seed)
+        {
+            var sha = new Sha256();
+            var result = BCryptOpenAlgorithmProvider(out sha._algorithm, "SHA256", null, BCryptFlags.BCRYPT_ALG_HANDLE_HMAC);
+            if (result != 0) throw new Exception();
```

Comment:
this is an internal class. If we ever productize it, we will cleanup the error code paths. In the prototyping phase, it's seems like not worth the effort.